### PR TITLE
Fix piecewise interpolate with 0 first split

### PR DESCRIPTION
**Pull request by rdyro**
Addresses https://github.com/google-deepmind/optax/issues/1403

### CODE REVIEW

File: optax/schedules/_schedule.py [hunk 1]
```diff
@@ -465,7 +465,7 @@ def piecewise_interpolate_schedule(
 
   bounds = np.stack((0,) + boundaries)
   values = np.cumprod(np.stack((init_value,) + scales))
-  interval_sizes = bounds[1:] - bounds[:-1]
+  interval_sizes = np.maximum(bounds[1:] - bounds[:-1], 1)
```
Comment (emilyfertig) [inline optax/schedules/_schedule.py:468]: Maybe I'm reading this wrong but is this just checking that the keys of boundaries_and_scales is unique? Would it be better to check that explicitly and raise if not?

File: optax/schedules/_schedule_test.py [hunk 2]
```diff
@@ -654,6 +654,13 @@ def test_invalid_scale(self):
     with self.assertRaises(ValueError):
       _schedule.piecewise_interpolate_schedule('linear', 13.0, {5: -3})
 
+  def test_zero_interval_size(self):
+    sched = _schedule.piecewise_interpolate_schedule(
+      'linear', 13.0, {0: 1.0, 5: 1.0}
+    )
+    if not all(jnp.isfinite(sched(i)) for i in range(6)):
+      raise ValueError('Piecewise schedule gives nan values for finite scales.')
```
Comment (emilyfertig) [inline optax/schedules/_schedule_test.py]: I'm confused by this test, should this be an assertion? (Also it looks like the test deals with non-unique scales, not bounds?)

File: optax/schedules/_schedule.py (hunk 1 above)
Comment (rdyro) [reply optax/schedules/_schedule.py:468]: Bounds are already passed as a dictionary with bounds as the keys, so they already have to be unique. This is a sanity check normalization for if the first bound is 0, resulting in 0 - 0 = 0

File: optax/schedules/_schedule_test.py (hunk 2 above)
Comment (rdyro) [reply optax/schedules/_schedule_test.py]: The message is confusing and this should be an assert, thanks, fiixed!

File: optax/schedules/_schedule.py (hunk 1 above)
Comment (emilyfertig) [reply optax/schedules/_schedule.py:468]: I see, thanks! Maybe consider adding a brief test that the interpolated values are as expected if 0 is in the keys.
